src/AutoMapper/TypeDetails.cs

```diff
@@ -138,77 +107,34 @@ select extensionMethod.MakeGenericMethod(genericInterfaceArguments)
-
-        private static MemberInfo[] BuildPublicAccessors(IEnumerable<MemberInfo> allMembers) =>
+        private MemberInfo[] BuildWriteAccessors() =>
             // Multiple types may define the same property (e.g. the class and multiple interfaces) - filter this to one of those properties
-            allMembers
-                .OfType<PropertyInfo>()
+            GetProperties(PropertyWritable)
                 .GroupBy(x => x.Name) // group properties of the same name together
                 .Select(x => x.FirstOrDefault(y => y.CanWrite && y.CanRead) ?? x.First()) // favor the first property that can both read & write - otherwise pick the first one
-                .Concat(allMembers.Where(x => x is FieldInfo)) // add FieldInfo objects back
+                .Concat(GetFields(FieldWritable))
                 .ToArray();
-
-        private IEnumerable<MemberInfo> GetAllPublicReadableMembers(Func<MemberInfo, bool> membersToMap)
-            => GetAllPublicMembers(PropertyReadable, FieldReadable, membersToMap);
-
-        private IEnumerable<MemberInfo> GetAllPublicWritableMembers(Func<MemberInfo, bool> membersToMap)
-            => GetAllPublicMembers(PropertyWritable, FieldWritable, membersToMap);
-
-        private IEnumerable<ConstructorInfo> GetAllConstructors(Func<ConstructorInfo, bool> shouldUseConstructor)
-            => Type.GetDeclaredConstructors().Where(shouldUseConstructor).ToArray();
-
         private static bool PropertyReadable(PropertyInfo propertyInfo) => propertyInfo.CanRead;
-
         private static bool FieldReadable(FieldInfo fieldInfo) => true;
-
         private static bool PropertyWritable(PropertyInfo propertyInfo) => propertyInfo.CanWrite || propertyInfo.PropertyType.IsNonStringEnumerable();
-
         private static bool FieldWritable(FieldInfo fieldInfo) => !fieldInfo.IsInitOnly;
-
-        private IEnumerable<MemberInfo> GetAllPublicMembers(
-            Func<PropertyInfo, bool> propertyAvailableFor,
-            Func<FieldInfo, bool> fieldAvailableFor,
-            Func<MemberInfo, bool> memberAvailableFor)
-        {
-            var typesToScan = new List<Type>();
-            for (var t = Type; t != null; t = t.BaseType)
-                typesToScan.Add(t);
-
-            if (Type.IsInterface)
-                typesToScan.AddRange(Type.GetTypeInfo().ImplementedInterfaces);
-
-            // Scan all types for public properties and fields
-            return typesToScan
-                .Where(x => x != null) // filter out null types (e.g. type.BaseType == null)
-                .SelectMany(x => x.GetDeclaredMembers()
-                    .Where(mi => mi.DeclaringType != null && mi.DeclaringType == x)
-                    .Where(
-                        m =>
-                            m is FieldInfo && fieldAvailableFor((FieldInfo)m) ||
-                            m is PropertyInfo && propertyAvailableFor((PropertyInfo)m) &&
-                            !((PropertyInfo)m).GetIndexParameters().Any())
```

Comment:
I removed this because it's expensive. I think it's rare and people can just explicitly ignore index properties.